app/questionnaire/rules.py

```diff
@@ -52,7 +57,13 @@ def evaluate_repeat(repeat_rule, answer_store):
         repeat_index = repeat_rule['answer_id']
         filtered = answer_store.filter(answer_id=repeat_index)
         repeat_function = repeat_functions[repeat_rule['type']]
-        return repeat_function(filtered)
+        no_of_repeats = repeat_function(filtered)
+
+        if no_of_repeats > 50:
```

Comment:
Would be good if this was configurable, ENV var at least so we can change the behaviour easily if it causes a problem as this feels like a system-wide imposed limit (like the maximum size of a POST data allowed etc.)